source/pick-from-possibly-undefined.d.ts

```diff
@@ -0,0 +1,19 @@
+/**
+It allows to pick properties from a type that may be undefined
+
+@example:
+type User = {
+    id: number;
+    name: string;
+    email: string;
+} | undefined;
+
+type UserWithId = PickFromPossiblyUndefined<User, 'id' | 'name'>;
+
+Results in: UserWithId = { id: number, name: string } | undefined
```

Comment:
Result seem is `{ id: number, name: string }`, not `{ id: number, name: string } | undefined`. 
Did this meet your expectations?

Reply:
Yes, sorry, you're right.